src/core/server/opensearch/version_check/ensure_opensearch_version.test.ts

```diff
@@ -356,6 +366,154 @@ describe('pollOpenSearchNodesVersion', () => {
       });
   });
 
+  it('returns compatibility results and isCompatible=true with filters and multiple cluster ids', (done) => {
+    expect.assertions(3);
+    const nodes = {} as any;
+    nodes['node-0'] = {
+      version: '5.1.0',
+      http: {
+        publish_address: 'http_address',
+      },
+      ip: 'ip',
+      attributes: {
+        cluster_id: '0',
+        custom_attribute: 'foo',
+      },
+    };
+    nodes['node-1'] = {
+      version: '5.1.0',
+      http: {
+        publish_address: 'http_address',
+      },
+      ip: 'ip',
+      attributes: {
+        cluster_id: '1',
+        custom_attribute: 'foo',
+      },
+    };
+    nodes['node-2'] = {
+      version: '6.2.0',
+      http: {
+        publish_address: 'http_address',
+      },
+      ip: 'ip',
+      attributes: {
+        cluster_id: '2',
+        custom_attribute: 'bar',
+      },
+    };
+    const filteredNodes = nodes;
+    delete filteredNodes['node-2'];
```

Comment:
What the purpose of this? when you delete from filteredNodes, you also delete from nodes since `filteredNodes` simply references `nodes`. if you want a copy, use the destructuring syntax

```ts
const filteredNodes = { ...nodes };
delete filteredNodes['node-2'];
```